core/Plugin/Manager.php

```diff
@@ -319,12 +319,14 @@ public function isPluginLoaded($name)
      */
     public function readPluginsDirectory()
     {
+        $isSystemValidationEnabled = SettingsPiwik::isSystemValidationEnabled();
```

Comment:
Same here as in https://github.com/matomo-org/matomo/pull/14624/files#r300292474

We know the plugin structures are valid and therefore we don't want to execute this on every request as it's a lot of stat calls.